npm/pkg/dataplane/dataplane_windows.go

```diff
@@ -347,31 +346,67 @@ func (dp *DataPlane) getEndpointsToApplyPolicies(netPols []*policies.NPMNetworkP
 
 func (dp *DataPlane) getLocalPodEndpoints() ([]*hcn.HostComputeEndpoint, error) {
 	klog.Info("getting local endpoints")
+
+	// Gets endpoints in state: Attached
 	timer := metrics.StartNewTimer()
-	endpoints, err := dp.ioShim.Hns.ListEndpointsQuery(dp.endpointQuery.query)
+	endpointsAttached, err := dp.ioShim.Hns.ListEndpointsQuery(dp.endpointQueryAttachedState.query)
 	metrics.RecordListEndpointsLatency(timer)
 	if err != nil {
 		metrics.IncListEndpointsFailures()
-		return nil, errors.Wrap(err, "failed to get local pod endpoints")
+		return nil, errors.Wrap(err, "failed to get local pod endpoints in state:attached")
 	}
 
-	if dp.EnableNPMLite {
-		timer = metrics.StartNewTimer()
-		endpointsAttached, errL1vh := dp.ioShim.Hns.ListEndpointsQuery(dp.endpointQueryL1VH.query)
-		metrics.RecordListEndpointsLatency(timer)
-		if errL1vh != nil {
-			metrics.IncListEndpointsFailures()
-			return nil, errors.Wrap(errL1vh, "failed to get local pod endpoints in L1VH")
-		}
-		endpoints = append(endpoints, endpointsAttached...)
+	// Gets endpoints in state: AttachedSharing
+	timer = metrics.StartNewTimer()
+	endpoints, err := dp.ioShim.Hns.ListEndpointsQuery(dp.endpointQuery.query)
+	metrics.RecordListEndpointsLatency(timer)
+	if err != nil {
+		metrics.IncListEndpointsFailures()
+		return nil, errors.Wrap(err, "failed to get local pod endpoints in state: attachedSharing")
 	}
+
+	// Filtering out any of the same endpoints between endpoints with state attached and attachedSharing
+	endpoints = removeCommonEndpoints(&endpoints, &endpointsAttached)
+
 	epPointers := make([]*hcn.HostComputeEndpoint, 0, len(endpoints))
 	for k := range endpoints {
 		epPointers = append(epPointers, &endpoints[k])
 	}
 	return epPointers, nil
 }
 
+func removeCommonEndpoints(endpoints, endpointsAttached *[]hcn.HostComputeEndpoint) []hcn.HostComputeEndpoint {
```

Comment:
The exact operation it appears to do is the [symmetric difference](https://en.wikipedia.org/wiki/Symmetric_difference) of the two sets of endpoints.

Reply:
Thanks Tim for the exact name. @huntergregory, changed the name to GetUniqueEndpoints. I chose this name as it may be easier to comprehend what the function is about. Please let me know if that works.